tools/cli/commands/rsync.js

```diff
@@ -68,17 +70,95 @@ export async function rsyncInit( argv ) {
 	}
 
 	if ( argv.watch ) {
-		console.log( 'jetpack rsync --watch is now watching for changes to:', sourcePluginPath );
-		const debouncedRsyncToDest = pDebounce(
-			() => rsyncToDest( sourcePluginPath, finalDest, argv.dest ),
-			10000 // Todo: at 10000ms, standard Jetpack build ran this twice for me, adjust further?
-		);
-		// eslint-disable-next-line no-unused-vars
-		chokidar.watch( sourcePluginPath ).on( 'change', async changedPath => {
-			await debouncedRsyncToDest();
-		} );
+		let watcher;
+		const rsyncAndUpdateWatches = async ( event, eventfile ) => {
+			if ( argv.v ) {
+				console.debug( `rsync due to event ${ event } for ${ eventfile }` );
+			}
+
+			const paths = await rsyncToDest( sourcePluginPath, finalDest );
+
+			if ( ! watcher ) {
+				watcher = chokidar.watch( [], {
+					cwd: sourcePluginPath,
+					followSymlinks: false,
+					disableGlobbing: true,
+					ignoreInitial: true,
+					depth: 0,
+				} );
+
+				// Always watch the plugin base dir.
+				watcher.add( '.' );
+
+				// Also watch the git index for changes to catch `git add`, as that may change which files are synced.
+				watcher.add( path.join( process.cwd(), '.git/index' ) );
+
+				// Watch `.gitignore` and `.gitattributes` in parent dirs, as they too may change which files are synced.
+				// Here we assume sourcePluginPath is always `projects/plugins/whatever`
+				watcher.add( '../.gitignore' );
+				watcher.add( '../.gitattributes' );
+				watcher.add( '../../.gitignore' );
+				watcher.add( '../../.gitattributes' );
+				watcher.add( '../../../.gitignore' );
+				watcher.add( '../../../.gitattributes' );
+
+				watcher.once( 'ready', () => {
+					console.log( 'jetpack rsync --watch is now watching for changes to:', sourcePluginPath );
+					watcher.on( 'all', pDebounce( rsyncAndUpdateWatches, 1000 ) );
+				} );
```

Comment:
As-is, I'm never seeing this block run 🤔 If I log out the watcher object it has readied up, so not quite sure what the mixup is here.

However, if I remove the `watcher.once` wrapper, then I can see the `rsync due to event change for` firing as expected. Since there is the `if ( ! watcher )` outside, do we need `once ready` here even?

Reply:
The intent of the wrapper is to try to avoid the events it generates on startup from triggering a new do-nothing rsync immediately, by not listening to anything before it signals 'ready'.

I wonder if 1bcb59ed85 will help on your system.

Reply:
After patching, still looks like I'm not getting inside that `watcher.once( 'ready'` bit. That's working on your end I take it?

Reply:
Yeah, works fine for me. Feel like trying a few other tests? From `tools/cli/`, try running these commands:
```
node -e 'const chokidar = require( "chokidar" ); const watcher = chokidar.watch( [ "." ] ); watcher.on( "ready", () => { console.log( "ready!" ); } );'
```
```
node -e 'const chokidar = require( "chokidar" ); const watcher = chokidar.watch( [] ); watcher.add( "." ); watcher.on( "ready", () => { console.log( "ready!" ); } );'
```
Both should print "ready!".

Reply:
Both of those printed `ready!` for me without issue.

Reply:
Huh. Try adding more stuff to bring it closer to what's in this PR until it breaks?
https://github.com/Automattic/jetpack/blob/1bcb59ed8553cb9074d396ed651d8662311e1786/tools/cli/commands/rsync.js#L82-L112
(keep in mind `process.cwd()` there is the monorepo root, but won't be with `node -e` from `tools/cli/`)

Reply:
Brad and I were able to figure out what was breaking, 4d78d76 should workaround the issue of the 'ready' event not firing I was seeing.